src/ray/util/memory.cc

```diff
@@ -47,18 +52,14 @@ void parallel_memcopy(uint8_t *dst, const uint8_t *src, int64_t nbytes,
 
   // Start all threads first and handle leftovers while threads run.
   for (int i = 0; i < num_threads; i++) {
-    threadpool[i] = std::thread(std::memcpy, dst + prefix + i * chunk_size,
-                                left + i * chunk_size, chunk_size);
+    boost::asio::post(threadpool, std::bind(std::memcpy, dst + prefix + i * chunk_size,
+                                            left + i * chunk_size, chunk_size));
```

Comment:
I think we should use std::future for synchronization here.
Besides, threadpool join seems not necessary.